src/AbstractWriter.ts

```diff
@@ -27,7 +27,13 @@ export abstract class AbstractWriter implements Writer {
     }
 
     setAnnotations(annotations: string[]): void {
-        this._annotations = annotations;
+        if (annotations === undefined || annotations === null) {
+            this._annotations = [];
+        } else if(!Array.isArray(annotations)){
+            throw new Error("Annotations must be an array of strings.")
+        } else {
+            this._annotations = annotations;
```

Comment:
don't you need to check the elements type as well? You are checking it in `IonBinaryWriter#encodeAnnotations`

Reply:
The elements are getting checked on write, the array is being checked when being set.
users are able to call writer.addAnnotation(string)

Reply:
It's always good to fail fast so it's easier for users to debug and find their mistake

Reply:
done